### PR TITLE
Add automatic retries on network errors and 503 responses

### PR DESCRIPTION
**Pull request by jimbobhickville**
I wasn't able to properly mock the guts of urllib3 in such
a way as to create unit tests for the retries, but I did
manual testing and it worked as expected.

### CODE REVIEW

File: lavaclient/_version.py [hunk 1]
```diff
@@ -10,5 +10,5 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
-__version_info__ = (0, 2, 8)
+__version_info__ = (0, 2, 9)
```
Comment (thesquelched) [inline lavaclient/_version.py]: 0.2.8 is unreleased on pypi, so just keep it there

File: lavaclient/cli.py [hunk 2]
```diff
@@ -76,6 +76,14 @@ def create_client(args):
                                   os.environ.get('LAVA2_API_URL'),
                                   os.environ.get('LAVA_API_URL')),
             verify_ssl=args.verify_ssl,
+            timeout=first_exists(args.request_timeout,
+                                 os.environ.get('LAVA_API_TIMEOUT')),
```
Comment (thesquelched) [inline lavaclient/cli.py]: Why not just `LAVA_TIMEOUT` and `LAVA_RETRIES`?

Comment (thesquelched) [inline lavaclient/cli.py]: The only thing that has `API` in it is the URL

File: lavaclient/_version.py (hunk 1 above)
Comment (jimbobhickville) [inline lavaclient/_version.py]: Ok, didn't realize.

File: lavaclient/cli.py (hunk 2 above)
Comment (jimbobhickville) [inline lavaclient/cli.py]: Fair enough.

**Comment by thesquelched** [comment]
A++++ would review again :+1:

**Comment by rnirmal** [comment]
+1

**Pull request merged**